### PR TITLE
DTD-2140-WIP  Added wip tag to some scenarios failing failng due to breathing space

### PR DESCRIPTION
**Pull request by thadeusssewanyana**

### CODE REVIEW

File: src/test/resources/features/ifs/DebtCalculationValidation.feature [hunk 1]
```diff
@@ -38,7 +38,7 @@ Feature: Debt Calculation Validation
 #      | originalAmount | dateCreated | interestStartDate | interestRequestedTo | mainTrans | subTrans | interestBearing |
 #      | 1.2            | 2021-03-01  | 2021-03-01        | 2021-03-08          | 1525      | 1000     | true            |
 #    And the debt item has no payment history
-#    And no breathing spaces have been applied to the customer
+#    And no breathing spaces have been applied to the debt item
```
Comment (vinniebrice) [inline src/test/resources/features/ifs/DebtCalculationValidation.feature:41]: This is all commented out

File: src/test/resources/features/ifs/MultipeDebtItems.feature [hunk 2]
```diff
@@ -178,3 +180,171 @@ Feature: Multiple Debt Items
...
+      | periodFrom | periodTo   | numberOfDays | interestRate | interestDueDailyAccrual | unpaidAmountWindow |
+      | 2018-12-16 | 2019-04-14 | 119          | 3.25         | 44                      | 505297             |
+
+
+  @wip
+  Scenario: Scenario 1 - Customer has 2 debts which are eligible for the Breathing Space protections.
+  A part payment is received post BS period & a further Interest Charge is raised.
+  IFS is called during BS period, after BS period for issue of an SOL and clerically following receipt of a payment.
+  Breathing space for interest bearing debt with payments.
+    Given a fc vat debt item
```
Comment (vinniebrice) [inline src/test/resources/features/ifs/MultipeDebtItems.feature]: Don't need to add FC scenario

File: src/test/resources/testdata/debtId.txt [hunk 3]
```diff
@@ -7,4 +7,3 @@
                 "interestIndicator": "<REPLACE_interestIndicator>",
                 "periodEnd": "<REPLACE_periodEnd>",
                 "paymentHistory": [<REPLACE_payments>]
-}
```
Comment (vinniebrice) [inline src/test/resources/testdata/debtId.txt]: Why was this removed?

File: src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/ifs/InterestForecastingSteps.scala [hunk 4]
```diff
@@ -416,11 +416,12 @@ class InterestForecastingSteps extends ScalaDsl with EN with Eventually with Mat
     responseBody should be(expectedMessage)
   }
 
-  Given("the customer has breathing spaces applied") { (dataTable: DataTable) =>
-    addBreathingSpace(dataTable)
+  Given("the debt item has breathing spaces applied") { (dataTable: DataTable) =>
+    addDebtBreathingSpace(dataTable)
+    //addBreathingSpace(dataTable)
```
Comment (vinniebrice) [inline src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/ifs/InterestForecastingSteps.scala]: Comment left in

File: src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala [hunk 5]
```diff
@@ -163,10 +166,9 @@ object InterestForecastingRequests extends ScalaDsl with EN with Eventually with
     // Set scenario Context to be all debt items with payments.
     ScenarioContext.set(
       "debtItems",
-      getBodyAsString("debtCalcRequest")
-        .replaceAll("<REPLACE_debtItems>", ScenarioContext.get("debtItems"))
+      getBodyAsString("debtCalcRequest").replaceAll("<REPLACE_debtItems>", ScenarioContext.get("debtItems"))
     )
-
+    //    ScenarioContext.set("debtItems",getBodyAsString("debtCalcRequest").replaceAll("<REPLACE_debtItems>", ScenarioContext.get("debtItems")))
```
Comment (vinniebrice) [inline src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala]: Comment left in

File: src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala [hunk 6]
```diff
@@ -186,27 +189,54 @@ object InterestForecastingRequests extends ScalaDsl with EN with Eventually with
...
-
     }
 
     val jsonWithbreathingSpaces =
       ScenarioContext.get("debtItems").toString.replaceAll("<REPLACE_breathingSpaces>", breathingSpaces)
     ScenarioContext.set("debtItems", jsonWithbreathingSpaces)
   }
 
+  def addDebtBreathingSpace(dataTable: DataTable): Unit = {
+    //ScenarioContext.set("debtItems",getBodyAsString("debtCalcRequest").replaceAll("<REPLACE_debtItems>", ScenarioContext.get("debtItems")))
```
Comment (vinniebrice) [inline src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala]: Comment left in

File: src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala
Comment (vinniebrice) [inline src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala:1]: Do we need `addBreathingSpace` and `addDebtBreathingSpace`

**Review by vinniebrice — CHANGES_REQUESTED** [review]
See comments

File: src/test/resources/features/ifs/DebtCalculationValidation.feature (hunk 1 above)
Comment (thadeusssewanyana) [reply src/test/resources/features/ifs/DebtCalculationValidation.feature:41]: commented out steps not part of my changes - one step defination changed

File: src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala
Comment (thadeusssewanyana) [reply src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala:1]: remove addDebtBreathingSpace

File: src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala (hunk 6 above)
Comment (thadeusssewanyana) [reply src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala]: method removed

File: src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala (hunk 5 above)
Comment (thadeusssewanyana) [reply src/test/scala/uk/gov/hmrc/test/api/requests/InterestForecastingRequests.scala]: removing

File: src/test/resources/features/ifs/MultipeDebtItems.feature (hunk 2 above)
Comment (vinniebrice) [reply src/test/resources/features/ifs/MultipeDebtItems.feature]: Done

**Review by vinniebrice — APPROVED** [review]
